### PR TITLE
Feature/retrolambda

### PR DESCRIPTION
**Pull request by wotomas**
- PR for #14 (Use retrolambda and add lambda expressions)

### CODE REVIEW

File: app/src/main/java/com/esoxjem/movieguide/details/MovieDetailsInteractorImpl.java [hunk 1]
```diff
@@ -30,54 +30,45 @@
     @Override
     public Observable<List<Video>> getTrailers(final String id)
     {
-        return Observable.defer(new Func0<Observable<List<Video>>>()
-        {
-            @Override
-            public Observable<List<Video>> call()
+        return Observable.defer(() -> {
```
Comment (esoxjem) [inline app/src/main/java/com/esoxjem/movieguide/details/MovieDetailsInteractorImpl.java]: Please use `Observable.fromCallable()` instead.

File: app/src/main/java/com/esoxjem/movieguide/details/MovieDetailsInteractorImpl.java [hunk 2]
```diff
@@ -30,54 +30,45 @@
...
+    }
+
     @Override
     public Observable<List<Review>> getReviews(final String id)
     {
-        return Observable.defer(new Func0<Observable<List<Review>>>()
-        {
-            @Override
-            public Observable<List<Review>> call()
+        return Observable.defer(() -> {
```
Comment (esoxjem) [inline app/src/main/java/com/esoxjem/movieguide/details/MovieDetailsInteractorImpl.java]: Same as above

File: app/src/main/java/com/esoxjem/movieguide/listing/MoviesListingInteractorImpl.java [hunk 3]
```diff
@@ -39,42 +39,36 @@
...
-            public Observable<List<Movie>> call()
-            {
-                try
-                {
-                    return Observable.just(get());
-                } catch (Exception e)
-                {
-                    return Observable.error(e);
-                }
+        return Observable.defer(() -> {
```
Comment (esoxjem) [inline app/src/main/java/com/esoxjem/movieguide/listing/MoviesListingInteractorImpl.java]: Same as above

File: app/src/main/java/com/esoxjem/movieguide/details/MovieDetailsInteractorImpl.java (hunk 1 above)
Comment (wotomas) [inline app/src/main/java/com/esoxjem/movieguide/details/MovieDetailsInteractorImpl.java]: Okay :)

**Comment by wotomas** [comment]
👍

**Pull request merged**